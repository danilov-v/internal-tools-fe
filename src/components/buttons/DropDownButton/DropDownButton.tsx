import React from 'react';

import { Button, ButtonProps } from 'components/buttons/Button';
import { ReactComponent as Up } from 'assets/icons/up.svg';
import { ReactComponent as Down } from 'assets/icons/down.svg';

const DropDownButton: React.FC<DropDownButtonProps> = ({
  expanded = false,
  ...rest
}) => (
  <Button
    {...rest}
    endIcon={
      expanded ? <Up data-testid="up-svg" /> : <Down data-testid="down-svg" />
    }
  />
);

interface DropDownButtonProps extends ButtonProps {
  expanded?: boolean;
}

export { DropDownButton };
