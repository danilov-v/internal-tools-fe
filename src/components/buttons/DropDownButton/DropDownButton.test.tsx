import React from 'react';
import { render, RenderResult } from '@testing-library/react';
import { DropDownButton } from './DropDownButton';

const getComponent = (props = {}): RenderResult => {
  const parsedProps = {
    ...props,
  };

  return render(<DropDownButton {...parsedProps} />);
};

describe('DropDownButton component', () => {
  it('Should render and match snapshot', () => {
    expect(getComponent().baseElement).toMatchSnapshot();
  });

  it('Should render Button with Down icon', () => {
    expect(getComponent().queryByTestId('down-svg')).toBeTruthy();
  });

  it('Should render Button with Up icon', () => {
    const component = getComponent({
      expanded: true,
    });

    expect(component.queryByTestId('up-svg')).toBeTruthy();
  });
});
