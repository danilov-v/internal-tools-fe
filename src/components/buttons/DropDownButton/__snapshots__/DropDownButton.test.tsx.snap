// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropDownButton component Should render and match snapshot 1`] = `
<body>
  <div>
    <button
      class="sc-AxjAm dGijUn"
      color="default"
      name=""
      type="button"
    >
      <span
        class="sc-AxirZ cXnwHz"
      />
      <svg
        data-testid="down-svg"
      >
        down.svg
      </svg>
    </button>
  </div>
</body>
`;
